File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/LoanCustomer.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class LoanCustomer : ApiBase
{
    [JsonProperty("firstName")] public string? FirstName { get; set; }


    [JsonProperty("lastName")] public string? LastName { get; set; }


    [JsonProperty("email")] public string? Email { get; set; }


    [JsonProperty("phone")] public string? Phone { get; set; }


    [JsonProperty("billingAddress")] public LoanAddress? BillingAddress { get; set; }


    [JsonProperty("shippingAddress")] public LoanAddress? ShippingAddress { get; set; }


    public string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, FirstName);
        Add(sb, LastName);
        Add(sb, Email);
        Add(sb, Phone);
        Add(sb, BillingAddress?.ToSign());
        Add(sb, ShippingAddress?.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/LoanCustomer.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class LoanCustomer extends ApiBase implements Signable {
    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private LoanAddress billingAddress;
    private LoanAddress shippingAddress;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, firstName);
        add(sb, lastName);
        add(sb, email);
        add(sb, phone);
        if(null != billingAddress) add(sb, billingAddress.toSign());
        if(null != shippingAddress) add(sb, shippingAddress.toSign());
        deleteLast(sb);
        return sb.toString();
    }
}
